domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/kafka/processor/StockPriceNotificationProcessor.kt

```diff
@@ -1,79 +1,36 @@
 package systems.ajax.malov.stockanalyzer.kafka.processor
 
 import jakarta.annotation.PostConstruct
-import org.apache.kafka.clients.consumer.ConsumerRecord
-import org.apache.kafka.clients.producer.ProducerRecord
-import org.bson.types.ObjectId
 import org.springframework.stereotype.Component
 import reactor.core.scheduler.Schedulers
 import reactor.kafka.receiver.KafkaReceiver
-import reactor.kafka.sender.KafkaSender
-import reactor.kafka.sender.SenderRecord
-import reactor.kotlin.core.publisher.toFlux
-import systems.ajax.malov.internalapi.KafkaTopic
-import systems.ajax.malov.internalapi.output.pubsub.stock.notification_stock_price.proto.NotificationStockPrice
+import reactor.kotlin.core.publisher.toMono
 import systems.ajax.malov.internalapi.output.pubsub.stock.stock_price.proto.StockPrice
-import systems.ajax.malov.stockanalyzer.entity.MongoUserTrackedSymbol
-import systems.ajax.malov.stockanalyzer.mapper.proto.BigDecimalProtoMapper.convertBigDecimalProtoToBigDecimal
-import systems.ajax.malov.stockanalyzer.repository.UserTrackedSymbolRepository
+import systems.ajax.malov.stockanalyzer.kafka.producer.StockPriceNotificationProducer
+import systems.ajax.malov.stockanalyzer.service.NotificationStockPriceService
 
 @Component
 class StockPriceNotificationProcessor(
     private val kafkaStockPriceConsumer: KafkaReceiver<String, StockPrice>,
-    private val kafkaNotificationStockPriceSender: KafkaSender<String, NotificationStockPrice>,
-    private val userTrackedSymbolRepository: UserTrackedSymbolRepository,
+    private val kafkaNotificationStockPriceSender: StockPriceNotificationProducer,
+    private val notificationStockPriceService: NotificationStockPriceService,
 ) {
     @PostConstruct
```

Comment:
what if the application context fails to start after creating your processor bean? is it correct that the not-ready application will consume traffic?

Reply:
Discussed it offline